pyro/poutine/runtime.py

```diff
@@ -24,26 +51,25 @@ class _DimAllocator:
     Note that dimensions are indexed from the right, e.g. -1, -2.
     """
 
-    def __init__(self):
-        self._stack = []  # in reverse orientation of log_prob.shape
+    def __init__(self) -> None:
+        # in reverse orientation of log_prob.shape
+        self._stack: List[Optional[str]] = []
 
-    def allocate(self, name, dim):
+    def allocate(self, name: str, dim: Optional[int]) -> int:
         """
         Allocate a dimension to an :class:`plate` with given name.
         Dim should be either None for automatic allocation or a negative
         integer for manual allocation.
         """
         if name in self._stack:
-            raise ValueError('duplicate plate "{}"'.format(name))
+            raise ValueError(f"duplicate plate '{name}'")
```

Comment:
No need to change, but I've been using the notation `f"duplicate plate {name!r}"` which I find easier to read.